rules/TypeDeclaration/TypeAnalyzer/GenericClassStringTypeNormalizer.php

```diff
@@ -41,6 +46,44 @@ public function normalize(Type $type): Type
 
             return $this->resolveStringType($value);
         });
+
+        if ($type instanceof UnionType) {
+            return $this->resolveClassStringInUnionType($type);
+        }
+
+        return $type;
+    }
+
+    private function resolveClassStringInUnionType(UnionType $type): Type
+    {
+        $unionTypes = $type->getTypes();
+        $isAllClassString = true;
+
+        foreach ($unionTypes as $unionType) {
+            if (! $unionType instanceof ArrayType) {
+                $isAllClassString = false;
+                break;
+            }
+
+            $keyType = $unionType->getKeyType();
+            $itemType = $unionType->getItemType();
+
+            if (! $keyType instanceof MixedType && ! $keyType instanceof ConstantIntegerType) {
+                $isAllClassString = false;
+                break;
+            }
+
+            if (! $itemType instanceof ClassStringType) {
+                $isAllClassString = false;
+                break;
+            }
+        }
```

Comment:
What about standalone method early return instead of `break`? 
`break` is somewhat confusing to read and here the `bool` value seems a good to return

Reply:
implemented https://github.com/rectorphp/rector-src/pull/518/commits/deb1a58680dada66b9b2414efc284d648e05c6a2

Reply:
Thank you :+1: